File: app/src/main/java/com/dominikgaller/alpinebooking/booking/core/domain/event/TourBookingRequested.java
package com.dominikgaller.alpinebooking.booking.core.domain.event;

import com.dominikgaller.alpinebooking.booking.core.domain.BookingId;
import com.dominikgaller.alpinebooking.booking.core.domain.ParticipantCount;
import com.dominikgaller.alpinebooking.booking.core.domain.TourDate;
import com.dominikgaller.alpinebooking.booking.core.domain.TourId;

import java.time.Instant;

/**
 * Domain event emitted when a new tour booking is requested.
 *
 * <p>Immutable. Published after successful transaction commit (see ADR 0002).
 *
 * <p>SDD: See {@code documentation/domain/aggregate-tour-booking.spec.md}, section 5.
 */
public record TourBookingRequested(
        BookingId bookingId,
        TourId tourId,
        TourDate tourDate,
        ParticipantCount participantCount,
        Instant occurredAt
) implements DomainEvent {
}

File: app/src/main/kotlin/com/example/service/booking/core/domain/tourbooking/event/TourBookingRequested.kt
package com.example.service.booking.core.domain.tourbooking.event

import com.example.service.booking.core.domain.tourbooking.BookingId
import com.example.service.booking.core.domain.tourbooking.ParticipantCount
import com.example.service.booking.core.domain.tourbooking.TourDate
import com.example.service.shared.domain.TourId
import com.example.service.shared.domain.event.DomainEvent
import java.time.Instant

/**
 * A booking was requested (UC01).
 *
 * SDD: see `documentation/use-cases/uc01-request-tour-booking.spec.md` § 6.
 */
data class TourBookingRequested(
    val bookingId: BookingId,
    val tourId: TourId,
    val tourDate: TourDate,
    val participantCount: ParticipantCount,
    val occurredAt: Instant,
) : DomainEvent
